src/transform.ts

```diff
@@ -63,3 +67,42 @@ export function processCss(src: string, filename: string): string {
 //   const baseCacheKey = cacheKeyFunction(src, filename, ...rest);
 //   return crypto.createHash('md5').update(baseCacheKey).digest('hex');
 // }
+
+// We cannot create async transformer if we are using CommonJS
+// ( Reference: https://github.com/facebook/jest/issues/11081#issuecomment-791259034
+// https://github.com/facebook/jest/issues/11458
+// Also, there is a inconsistency in jest docs about should `process` be required
+// https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object
+// A transformer must be an object with at least a process function
+// https://jestjs.io/docs/code-transformation#writing-custom-transformers
+// As can be seen, only process or processAsync is mandatory to implement)
+
+// We can use that if we opt-in to ESM. But it's not common use case right now (2022)
+// So our approach is making CSS Modules a "CSS-in-JS" solution.
+// CSS Modules will be compiled at run time then inject to the document.body
+// One notable note is that `postcss-modules` is an async postcss plugin
+// so we need to use `postcss-modules.sync`, with function `sync()`
+function processCSSModules(src: string, filename: string): string {
+  return `
+const postcss = require('postcss');
+const CSSModulesSync = require('postcss-modules-sync').default;
+const cssSrc = ${JSON.stringify(src)};
+
+let exportedTokens = {};
+
+const result = postcss(
+  CSSModulesSync({
+    getJSON: (tokens) => {
+      exportedTokens = tokens;
+    },
+  }),
+)
+.process(cssSrc, { from: ${JSON.stringify(filename)} })
+
+const style = document.createElement('style');
+style.type = 'text/css';
+style.appendChild(document.createTextNode(result.css));
+document.body.appendChild(style);
+
+module.exports = exportedTokens`;
```

Comment:
I think creating an indent here would be better and consistent

Reply:
This is intentional since transformed code shouldn't have an indentation at every line.
(However, in the first line we there is a redundant blank line, I will remove this)

Reply:
We don't have consistency across transform functions. Let me make them consistent. Thanks